### DISCUSSION TITLE
Analysis Engine - 4. Feature List

### DISCUSSION THREAD

**Discussion by ansrlm** [githru/githru-vscode-ext#12] [General]
### 목표
- analysis engine의 feature list에 대해서 각자 생각하고 있는 부분들을 공유합니다.
- feature에 대한 세부사항을 포함해서 항목들을 확정하고 등록합니다.

### 참고사항
- feature list
  - git log
  - DAG
  - STEM
  - CSM
  - User Info

**Comment by anpaul0615** [comment]
feature list 에 대한 대략적인 구현내용(?) 을 정리해봤습니다.

처음엔 잘 몰랐었는데..  알고보니깐 engine 에서 다루는 feature 들이 
git-metadata-preprocessor 와 githru FE 일부분에서 가져와야 되겠더라구요.

- https://github.com/githru/git-metadata-preprocessor/blob/master/code/logGenerator.ipynb
- https://github.com/githru/githru/blob/master/frontend/src/components/CommitGraph.js
- https://github.com/githru/githru/blob/master/frontend/src/components/DetailCompare.js
- https://github.com/githru/githru/blob/master/frontend/src/components/BranchSummary.js
- ...


솔직히 아직까지는 위 코드들을 자세히 보지는 못했지만...ㅠㅠ
(아주아주) 대략적으로만 확인해서 간략하게 구현방향 정리해봤습니다.

- git log
  => git log 출력 파싱 + commits.json 생성

- DAG
  => parent커밋 연결 자료구조 생성 (+ commits.dag[n].json 생성..? or in-memory..?)
  ※ 추후 STEM 생성에서 timestamp 역방향 순회 필요
  ※ 추후 CSM 생성에서 최초커밋발 순회 필요

- STEM
  => 최근~과거 parent 역추적 + 커밋을 한줄덩어리(?) 로 구분

- CSM
  => 과거~최근 merge 커밋 감지 + 머지커밋이 참조하는 커밋들을 두꺼운커밋(?) 하나로 합치기

- User Info
  => git log 출력 내 author 정보와 github 프로필 연결 (key: 이메일)

**Comment by ansrlm** [comment]
- base
  - 각 피쳐의 타게팅이 되는 파이썬 파일을 보고 이해한 부분을 readme.md 파일을 작성해 공유하는 것 검토 필요
- git log
  - 해당 피쳐에서 scheme 정의까지 함께 하는 것은 괜찮을지 검토 필요
  - https://github.com/githru/githru-vscode-ext/pull/13
- DAG
  - STEM 생성을 하려면 그래프를 들고 있어야해서 runtime에 모두 처리할 수 있을지 검토 필요
  - edge case (커밋수가 많다거나,,,) 한 부분 제외

**Reply by anpaul0615** [reply]
>  readme.md 파일을 작성해 공유하는 것 검토 필요

각 feature 의 구현과 더불어 해당 feature 의 input/output 및 interface 정의 정리해서 readme 작성
=> PR1 : feature 구현
=> PR2 : feature 구현에 대한 readme

**Comment by anpaul0615** [comment]
features
- git log => 효진
- DAG => 예진 (고민중)
- STEM => 바울
- CSM => 문기
- User Info (남는사람)

**Reply by anpaul0615** [reply]
담당 feature 에 대한 분석 정리 (위키에 정리)
=> 금주 토요일 회의에서 공유 (13시~, 약 2시간 진행)
=> 다음회의 일정 잡기

**Reply by hy57in** [reply]
각자 맡은 파트 조사해서 위키에 작성하기
